source/core_mqtt.c

```diff
@@ -737,10 +748,14 @@ static int32_t recvExact( const MQTTContext_t * pContext,
                         ( unsigned long ) bytesRemaining,
                         ( long int ) totalBytesRecvd ) );
         }
+        else
+        {
+            /* No bytes were read from the network. */
+            timeSinceLastDataWasRecvdMs = calculateElapsedTime( getTimeStampMs(), lastDataRecvTimeMs );
```

Comment:
I think expecting that a return value of 0 means no data to receive needs to be documented more clearly.

In POSIX sockets and OpenSSL, for example, a return value of 0 means the server has closed the connection or that a close-notify has been sent. No data to receive will actually return a negative error code. In our transport implementations, that error code is mapped appropriately to 0, but this is not going to be immediately obvious to other engineers who try to make their own transport implementations.

Reply:
Good point, I have updated that in the transport interface API doc.